nexus/src/app/mod.rs

```diff
@@ -58,6 +61,44 @@ pub(crate) const MAX_NICS_PER_INSTANCE: usize = 8;
 // TODO-completness: Support multiple external IPs
 pub(crate) const MAX_EXTERNAL_IPS_PER_INSTANCE: usize = 1;
 
+pub(crate) struct ExternalServers {
+    config: dropshot::ConfigDropshot,
+    https_port: u16,
+    http: Option<DropshotServer>,
+    https: Option<DropshotServer>,
+}
+
+impl ExternalServers {
+    pub fn new(config: dropshot::ConfigDropshot, https_port: u16) -> Self {
+        Self { config, https_port, http: None, https: None }
+    }
+
+    pub fn set_http(&mut self, http: DropshotServer) {
+        self.http = Some(http);
+    }
+
+    pub fn https_port(&self) -> u16 {
+        self.https_port
+    }
+
+    /// Returns a context object, if one exists.
+    pub fn get_context(&self) -> Option<Arc<crate::ServerContext>> {
+        if let Some(context) =
+            self.https.as_ref().map(|server| server.app_private())
+        {
+            // If an HTTPS server is already running, use that server context.
+            Some(context.clone())
+        } else if let Some(context) =
+            self.http.as_ref().map(|server| server.app_private())
+        {
+            // If an HTTP server is already running, use that server context.
+            Some(context.clone())
+        } else {
+            None
+        }
+    }
+}
```

Comment:
This feels a little weird to me. I guess it's because there's really just one `Arc<crate::ServerContext>` that's shared across http and https and we want to return it? Maybe it makes sense to just store a (non-optional) `Arc<crate::ServerContext>` within `ExternalServers` itself?

Is it guaranteed that there's always just one server context?